system/Images/Handlers/ImageMagickHandler.php

```diff
@@ -288,23 +288,34 @@ public function save(string $target = null, int $quality = 90): bool
 	{
 		$target = empty($target) ? $this->image() : $target;
 
-		// If no new resource has been created, then we're
-		// simply copy the existing one.
-		if (empty($this->resource))
+		// If no new resource has been created,  and we
+		// aren't adjusting the quality then we're simply
+		// copy the existing one.
+		if (empty($this->resource) && $quality == 100)
 		{
+			// we're moving it to the same spot so don't do anything.
+			if($target === null)
```

Comment:
I can't see the scenario when this condition would be used.

Reply:
This would fail for instance if someone set it to save to the same location and set a quality to 100. I'd think it would be easier for this to just leave it alone, than to fail trying to copy it to the same spot

Reply:
I understand but look at line 289. `$target` will never be a `null` value.